### PR TITLE
[WC-38] - Implement Course Detail

### PR DESCRIPTION
**Pull request by sikares**
# Description
This pull request introduces the creation of the Course Detail Page, which consists of several components: `CourseDetailFirstPage`, `MeetChefCurvePage`, `IngredientPage`, `JoinOurCoursePage`, `IngredientPage`, `client-review.ts`,`ingredient.ts`, ` reviewer.ts`, and `FAQPage`.

[//]: # 'Please include a summary of the changes and the related issue. (You can write below without deleting this comment)'

### What

[//]: # 'Briefly describe this feature / fix / refactor (You can write below without deleting this comment)'
This pull request introduces the Course Detail Page, which enhances user engagement by providing a comprehensive overview of the course. The page includes multiple components, such as CourseDetailFirstPage for an initial introduction, MeetChefCurvePage to showcase the chef's expertise, IngredientPage to highlight the ingredients used, JoinOurCoursePage for easy enrollment, and FAQPage to address common inquiries. Additionally, client-review.ts and reviewer.ts are implemented to manage and display participant reviews, building trust and credibility for the course offering.
image
image
image
image

### Why

[//]: # 'Describe the reason why we need this feature / fix / refactor or refer to Jira Ticket (You can write below without deleting this comment)'
This feature is essential to improve the user experience and engagement on the platform by providing a detailed and organized view of the course offerings. By creating a dedicated Course Detail Page, users can easily access crucial information, such as the chef's background, the ingredients used, and enrollment options. This enhances transparency and helps potential participants make informed decisions. Additionally, the inclusion of client reviews builds trust and credibility, which are vital for attracting new users. Implementing this feature aligns with our goal of delivering a comprehensive educational experience, ultimately leading to higher enrollment rates and customer satisfaction.

## Type of change

Please check the relevent option with `x`.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (if needed)
- [x] My changes generate no new warnings
- [x] I have tested on my `Local Machine`

## Release Notes

- **New Features**
	- Introduced multiple new components for enhanced course presentation, including:
		- `CourseDetailFirstPage`: Detailed view for a cooking course.
		- `FAQPage`: Interactive FAQ section for user inquiries.
		- `IngredientPage`: List of ingredients for the cooking course.
		- `JoinOurCoursePage`: Course enrollment statistics and user reviews.
		- `MeetChefCurvePage`: Introduction to Chef Sieng and his expertise.
		- `CourseDetailPage`: Composite layout for the course detail page.
- **Improvements**
	- Enhanced navigation structure and search functionality in the `NavbarAfter` and `NavbarBefore` components.
	- Adjusted layout and styling for better user experience across components.
	- Increased vertical padding in the `ClientReview` component for improved spacing.
	- Updated image dimensions and layout in the `OrderIngredientPage` for better visual appeal.

### CODE REVIEW

File: app/client/components/JoinOurCoursePage.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            <UserGroupIcon className="w-7 h-7 text-[#FE3511] mr-4" />
+            <div>
+              <div className="text-2xl font-bold">
+                {stats.enrolled / 1000}K+
+              </div>
+              <div className="text-gray-500">Enrolled</div>
+            </div>
+          </div>
+        </div>
+      </div>
```
[the remark this answers is not in the document]
Comment (sikares) [reply app/client/components/JoinOurCoursePage.tsx:99]: Code will error

File: app/client/components/JoinOurCoursePage.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                height={40}
+              />
+              <div>
+                <div className="font-semibold">{review.name}</div>
+                <div className="flex">{renderStars(review.rating)}</div>
+              </div>
+            </div>
+            <p className="text-gray-700">&quot;{review.comment}&quot;</p>
+          </div>
+        ))}
```
[the remark this answers is not in the document]
Comment (sikares) [reply app/client/components/JoinOurCoursePage.tsx:125]: Code will error

File: app/client/video/course-detail/[course_id]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+'use client'
+import NavbarAfter from '@/app/client/components/NavbarAfter'
+import CourseDetailFirstPage from '@/app/client/components/CourseDetailFirstPage'
+import MeetChefCurvePage from '@/app/client/components/MeetChefCurvePage'
+import IngredientPage from '@/app/client/components/IngredientPage'
+import JoinOurCoursePage from '@/app/client/components/JoinOurCoursePage'
+import FAQPage from '@/app/client/components/FAQPage'
+import Footer from '@/app/client/components/Footer'
```
[the remark this answers is not in the document]
Comment (sikares) [reply app/client/video/course-detail/[course_id]/page.tsx:8]: Not necessary to use useEffect and useState

File: app/client/video/course-detail/[course_id]/page.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      <NavbarAfter />
+      <CourseDetailFirstPage />
+      <MeetChefCurvePage />
+      <IngredientPage />
+      <JoinOurCoursePage />
+      <FAQPage />
+      <Footer />
+    </div>
+  )
+}
```
[the remark this answers is not in the document]
Comment (sikares) [reply app/client/video/course-detail/[course_id]/page.tsx:22]: I will change function name to CourseDetailPage

File: app/client/data/ingredient.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+export interface Ingredient {
+  name: string
+}
```
[the remark this answers is not in the document]
Comment (sikares) [reply app/client/data/ingredient.ts:3]: I will fix it

File: app/client/video/course-detail/[course_id]/page.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
```
[the remark this answers is not in the document]
Comment (sikares) [reply app/client/video/course-detail/[course_id]/page.tsx:21]: Code will error

File: app/client/components/IngredientPage.tsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
+import React from 'react'
+import Image from 'next/image'
+import { ingredients } from '../data/ingredient';
+
+export default function IngredientPage() {
+  return (
+    <div className="h-[450px] relative w-full flex flex-col items-center">
+      <div className="absolute left-0 top-[-140px] z-10 w-1/2 md:w-auto">
```
[the remark this answers is not in the document]
Comment (sikares) [reply app/client/components/IngredientPage.tsx:8]: I can't use

File: app/client/components/IngredientPage.tsx [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+        <Image
+          src="/images/spice.png"
+          alt="Spices"
+          width={750}
+          height={750}
+          className="object-contain w-full h-auto"
+        />
```
[the remark this answers is not in the document]
Comment (sikares) [reply app/client/components/IngredientPage.tsx:15]: I will fix it.

File: app/client/data/ingredient.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+  category?: string // e.g., 'dairy', 'meat', 'pasta'
+}
+
+export const ingredients: Ingredient[] = [
+  { name: 'Spaghetti' },
+  { name: 'Eggs' },
+  { name: 'Pecorino Romano Cheese' },
+  { name: 'Guanciale Black' },
+  { name: 'Pepper' },
+]
```
[the remark this answers is not in the document]
Comment (sikares) [reply app/client/data/ingredient.ts:15]: Thank, I will add it later.

**Comment by tong788** [comment]
Why don't you want to make a background square as shown in figma?
yours
image
In figma
image

**Comment by tong788** [comment]
image
Can you please change the name to WeCooked?

**Review by tong788 — APPROVED** [review]
Overview is ok in my opinion, just consider my 2 comments that I have published.

**Pull request merged**